File: src/server/src/main/java/com/project/domain/comment/Dislike.java
package com.project.domain.comment;

import com.project.domain.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

import java.util.UUID;

@Entity(name = "dislikes")
public class Dislike extends BaseEntity {
    @Id
    private UUID id;

    @ManyToOne
    private Comment comment;

    protected Dislike() {
    }

    public Dislike(Comment comment) {
        super();
        this.comment = comment;
    }
}

File: src/server/src/main/kotlin/com/project/posts/domain/Dislike.kt
package com.project.posts.domain

import com.project.common.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import java.util.UUID

@Entity(name = "dislikes")
class Dislike protected constructor() : BaseEntity() {
    @Id
    val id: UUID? = UUID.randomUUID()

    @ManyToOne
    var comment: Comment? = null

    constructor(comment: Comment) : this() {
        this.comment = comment
    }
}
